scripts/train/train.py

```diff
@@ -579,13 +548,27 @@ def main(cfg: DictConfig) -> Trainer:
     optimizer_name: str = optimizer_config.pop('name')
     optimizer = build_optimizer(model, optimizer_name, optimizer_config)
 
-    # Now add the eval metrics
+    ## Evaluation
+    print('Building eval loader...')
```

Comment:
this was actually written intentionally with creating the dataset before creating the model to help fast fail on dataset errors, since downloading the model can take a while (hence the weird structure of adding the metrics after the fact). I'd like to preserve that fast fail.

Reply:
This makes it a little annoying, but yea this makes sense 1050deff296484c93c9a2416eee96862e6394951